samples/bot-all-cards/csharp/README.md

```diff
@@ -0,0 +1,156 @@
+> the Teams service needs to call into the bot.
+
+1. Register a new application in the [Azure Active Directory – App Registrations](https://go.microsoft.com/fwlink/?linkid=2083908) portal.
+
+**NOTE:** When you create your bot you will create an App ID and App password - make sure you keep these for later.
+
+2. Setup for Bot
+- In Azure portal, create a [Azure Bot resource](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=csharp%2Caadv2).
+- Ensure that you've [enabled the Teams Channel](https://docs.microsoft.com/en-us/azure/bot-service/channel-connect-teams?view=azure-bot-service-4.0)
+- While registering the bot, use `https://<your_ngrok_url>/api/messages` as the messaging endpoint.
+
+3. Setup NGROK 
+1) Run ngrok - point to port 3978
+
+    ```bash
+    # ngrok http -host-header=rewrite 3978
+    ```
+4. Setup for code 
+- Clone the repository
+
+    ```bash
+    git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+    ```
+ Run the bot from a terminal or from Visual Studio:
+
+  A) From a terminal, navigate to `samples/bot-all-cards/csharp/BotAllCards`
+
+  ```bash
+  # run the bot
+  dotnet run
+  ```
+
+  B) Or from Visual Studio
+
+  - Launch Visual Studio
+  - File -> Open -> Project/Solution
+  - Navigate to `samples/bot-all-cards/csharp/BotAllCards` folder
+  - Select `BotAllCards.csproj` file
+  - Press `F5` to run the project   
+
+### 4. Setup Manifest for Teams
+
+- **This step is specific to Teams.**
+
+1) Modify the `manifest.json` in the `/Manifest` folder and replace the following details:
+  - `{{Microsoft-App-Id}}` with Application id generated from Step 1
+  - `{{domain-name}}` with base Url domain. E.g. if you are using ngrok it would be `1234.ngrok.io`
+
+2) Zip the contents of `Manifest` folder into a `manifest.zip`.
+
+3) Modify the `/appsettings.json` and fill in the following details:
+  - `{{Microsoft-App-Id}}` - Generated from Step 1 is the application app id
+  - `{{ Microsoft-App-Password}}` - Generated from Step 1, also referred to as Client secret
+  - `{{ Microsoft-App-TenantId }}` - Generated from Step 1 is the tenantId id
+  
+5) Upload the manifest.zip to Teams (in the Apps view click "Upload a custom app")
+   - Go to Microsoft Teams. From the lower left corner, select Apps
+   - From the lower left corner, choose Upload a custom App
+   - Go to your project directory, the ./Manifest folder, select the zip folder, and choose Open.
+   - Select Add in the pop-up dialog box. Your app is uploaded to Teams.
+
+## Running the sample
+
+**Install App:**
+
+![Installapp](BotAllCards/Images/Installapp.png)
+
+**Welcome Cards:**
+
+![WelcomeCards](BotAllCards/Images/WelcomeCards.png)
+
+**All Cards:**
+
+![AllCards](BotAllCards/Images/AllCards.png)
+
+**Adaptive Card:**
+
+![AdaptiveCard](BotAllCards/Images/AdaptiveCard.png)
+
+**Hero Card:**
+
+![HeroCard](BotAllCards/Images/HeroCard.png)
+
+**OAuth Card:**
```

Comment:
This should come after cloning the code or with configure bot connection section.

Reply:
Fixed